process/smartContract/builtInFunctions/keyValueStorage.go

```diff
@@ -37,58 +37,76 @@ func (k *saveKeyValueStorage) ProcessBuiltinFunction(
 	_, acntDst state.UserAccountHandler,
 	input *vmcommon.ContractCallInput,
 ) (*vmcommon.VMOutput, error) {
+	err := checkArgumentsForSaveKeyValue(acntDst, input)
+	if err != nil {
+		return nil, err
+	}
+
+	vmOutput := &vmcommon.VMOutput{
+		GasRemaining: input.GasProvided,
+	}
+
+	useGas := k.funcGasCost
+	for i := 0; i < len(input.Arguments); i += 2 {
+		key := input.Arguments[i]
+		value := input.Arguments[i+1]
+		length := uint64(len(value))
+		useGas += length * (k.gasConfig.DataCopyPerByte + k.gasConfig.PersistPerByte)
+
+		if !process.IsAllowedToSaveUnderKey(key) {
+			return nil, fmt.Errorf("%w it is not allowed to save under key %s", process.ErrOperationNotPermitted, key)
+		}
+
+		oldValue, _ := acntDst.DataTrieTracker().RetrieveValue(key)
```

Comment:
Should we treat error?

Reply:
no error is expected here - anyway it will be rewritten with the new value.